app/scripts/controllers/reviewRegistration.js

```diff
@@ -280,4 +280,158 @@ angular.module('confRegistrationWebApp')
+    var queryParameters = {
+      block: [],
+      //page: 1,
+      //limit: 20,
+      orderBy: 'last_name',
+      order: 'ASC',
+      filter: '',
+      filterPayment: '',
+      filterRegType: '',
+      includeCheckedin: 'yes',
+      includeWithdrawn: 'yes',
+      includeIncomplete: 'yes'
+    };
+
+    //Get all current registrants based on conference.id and queryParameters
+    //then check if spouse has registered and display 'Spouse Registration' section if needed.
+    var checkSpouseRegistration = function() { // jshint ignore:line
+      $http.get('conferences/' + conference.id + '/registrations', {params: queryParameters})
+        .then(function () {
+          var spouseIncludedOnCurrentRegistration = _.contains(_.map($scope.spouseRegistration.registrants, 'email'), $scope.currentRegistration.registrants[0].email);
+          // 2. If the registrant does have a spouse who has already registered, then
+          if ($scope.spouseRegistration.completed && !$scope.includeSpouseInRegistrationClicked) {
+            // A. If spouse is not included in that registration, then show another separate panel on the confirmation page,
+            //    indicating that their spouse has already registered and would they like to add their registration to their spouses?
+            //    . Call endpoint to copy registrant from one registration to another
+            //    . (possibly) Call updateRegistration
+            if (!spouseIncludedOnCurrentRegistration) {
+              $scope.spouseSectionVisible = true;
+            }
+            // B. Otherwise, indicate that they have already been included on their spouses registration. And possibly let them register anyway if they must.
+            else {
+              $scope.spouseSectionVisible = false;
+            }
+          }
+        }).catch(function(response){
+        console.log('Failed to get all registered users for current conference. Status = ' + response.status);
+      });
+    };
+
+    //Click event for 'Include Spouse' button
+    $scope.includeSpouseInRegistration = function () {
+
+      //Generate new local UUID used for registrantId
+      var _newRegistrantId = uuid();
+
+      //Payload for new spouse registration
+      var _newSpouseRegistration =
+        {
+          id: $scope.spouseRegistration.id,
+          conferenceId: $scope.currentRegistration.conferenceId,
+          registrants: [{
+            id: _newRegistrantId,
+            registrationId: $scope.spouseRegistration.id,
+            registrantTypeId: $scope.currentRegistration.registrants[0].registrantTypeId,
+            answers: []
+          }]
+        };
+
+      var _answers = '';
+
+      //Get all answers
+      _.forEach($scope.currentRegistration.registrants[0].answers, function(value) {
+        if (typeof(value.value) === 'object') {
+          value.value = null;
+        }
+        _answers = _answers +
+          '{' +
```

Comment:
Are you familiar with lodash's [`_.assign`](https://lodash.com/docs/2#assign) method? It allows you to copy properties from one object to another. So here we could do:

```js
var _answer = _.assign({}, oldAnswer, {
  id: uuid(),
  registrantId: _newRegistrantId
});
```

This starts with an empty object, copies all of the properties from `oldAnswer` into that object, and then overwrites `id` and `registrantId`. Doing that satisfies the "any other fields from the answer must be copied as well" requirement.